cpp/include/rmm_utils.h

```diff
@@ -37,6 +37,8 @@
   RMM_TRY_THROW( RMM_REALLOC((ptr), (sz), (stream)) ) \
 }
 
-#define ALLOC_FREE_TRY(ptr, stream){            \
-  RMM_TRY_THROW( RMM_FREE( (ptr), (stream) ) )  \
+#define ALLOC_FREE_TRY(ptr, stream){              \
+  if(rmmIsInitialized((rmmOptions_t*) NULL)) {    \
```

Comment:
Could you just add comments stating this is a temporary solution for this release and this needs to be revisited as we finalize our longer term strategy to address this issue.

Reply:
Yes, good idea, will do.